packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -226,35 +226,34 @@ async function _getDashboardStats(app) {
   }
 
   try {
-    const remotes = await app.getAllRemotes()
-    const remotesInfo = await app.getAllRemotesInfo()
+    const s3Brsize = { backups: 0 }
+    const otherBrSize = { available: 0, backups: 0, other: 0, total: 0, used: 0 }
 
-    const backupRepositoriesSize = remotes.reduce(
-      (prev, remote) => {
-        const { type } = parse(remote.url)
-        const remoteInfo = remotesInfo[remote.id]
+    const backupRepositories = await app.getAllRemotes()
+    const backupRepositoriesInfo = await app.getAllRemotesInfo()
 
-        if (!remote.enabled || type === 's3' || remoteInfo === undefined) {
-          return prev
-        }
+    for (const backupRepository of backupRepositories) {
+      const { type } = parse(backupRepository.url)
+      const backupRepositoryInfo = backupRepositoriesInfo[backupRepository.id]
 
-        return {
-          available: prev.available + remoteInfo.available,
-          backups: 0, // @TODO: compute the space used by backups
-          other: 0, // @TODO: compute the space used by everything that is not a backup
-          total: prev.total + remoteInfo.size,
-          used: prev.used + remoteInfo.used,
-        }
-      },
-      {
-        available: 0,
-        backups: 0,
-        other: 0,
-        total: 0,
-        used: 0,
+      if (!backupRepository.enabled || backupRepositoryInfo === undefined) {
+        continue
       }
-    )
-    dashboard.backupRepositories = { size: backupRepositoriesSize }
+
+      const { available, size, sizeUsedForBackups = 0, used } = backupRepositoryInfo
+      const isS3 = type === 's3'
+      const target = isS3 ? s3Brsize : otherBrSize
+
+      target.backups += sizeUsedForBackups
+      if (!isS3) {
```

Comment:
I think the isS3 test should be contained in RermoteAdapter, to hide/centralize  the discrepancy between storages in a future proof way

Ideally I would have the fs handler throw an "NOT IMPLEMENTED" exception caught by RemoteAdapter if possible

Reply:
after discussion, since the front UX is different when using objact storage and other storage, we can make the different computations here